arch/src/x86_64/mod.rs

```diff
@@ -145,6 +148,15 @@ struct BootParamsWrapper(boot_params);
 // SAFETY: BootParamsWrap is a wrapper over `boot_params` (a series of ints).
 unsafe impl ByteValued for BootParamsWrapper {}
 
+pub struct CpuidConfig {
+    pub sgx_epc_sections: Option<Vec<SgxEpcSection>>,
+    pub phys_bits: u8,
+    pub kvm_hyperv: bool,
+    #[cfg(feature = "tdx")]
+    pub tdx_enabled: bool,
```

Comment:
I'm not sure I like the `_enabled` suffix - it's a bool - it's obviously about enabling that functionality or not (and you didn't do it for `kvm_hyperv` so it's inconsistent.